src/backend/kvm/builder.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::config::Config;
+
+use std::convert::TryFrom;
+use std::sync::{Arc, RwLock};
+
+use crate::backend::kvm::mem::Region;
+use anyhow::{Error, Result};
+use kvm_bindings::bindings::kvm_userspace_memory_region;
+use kvm_bindings::fam_wrappers::KVM_MAX_CPUID_ENTRIES;
+use kvm_ioctls::{Kvm, VmFd};
+use lset::Span;
+use mmarinus::{perms, Map};
+use sallyport::elf::pf::kvm::SALLYPORT;
+use sallyport::Block;
+use std::collections::VecDeque;
+use std::mem::size_of;
+use std::num::NonZeroUsize;
+use x86_64::VirtAddr;
+
+pub struct Builder {
+    _cnfg: Config,
+    kvm_fd: Kvm,
+    vm_fd: VmFd,
+    regions: Vec<Region>,
+    sallyport_blocks: Option<Span<VirtAddr, NonZeroUsize>>,
```

Comment:
Why not `sallyports: Vec<VirtAddr>` ?

Reply:
Doing it this way forces sallyports to be contiguous.

Reply:
yes